resource/errors.go

```diff
@@ -68,6 +68,11 @@ func (e *mustRebuildError) Error() string {
 	return fmt.Sprintf("cannot reconfigure %q; must rebuild", e.name)
 }
 
+// NewBuildTimeoutError is used when a resource times out during construction or reconfiguration.
+func NewBuildTimeoutError(name Name) error {
+	return fmt.Errorf("resource %s timed out during reconfigure, update ResourceConfigurationTimeout to override", name)
```

Comment:
```suggestion
	return fmt.Errorf("resource %s timed out after %v during reconfigure", name, robotimpl.ResourceConfigurationTimeout)
```
I think, for now, `ResourceConfigurationTimeout` should not be a knob that we present to users. While a user _could_ alter `robotimpl.ResourceConfigurationTimeout` themselves, it's not a very idiomatic pattern and I don't think we should suggest it. If we _really_ want an escape hatch, I'd suggest something like an env var.

Reply:
🤦 yes you're absolutely right, in fact the scope doc even says it should be an env var! will update.